From d684055b2b92e7ec5793e70c9a80c7f8e45e0696 Mon Sep 17 00:00:00 2001
From: Bernd Kuhls <bernd.kuhls@t-online.de>
Date: Fri, 24 Aug 2018 16:38:42 +0200
Subject: [PATCH] _GNU_SOURCE needed for EAI_INPROGRESS
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Define of _GNU_SOURCE is needed to be able to use EAI_INPROGRESS in
loop.c.

This patch fixes a build error

loop.c:334:17: error: ‘EAI_INPROGRESS’ undeclared (first use in this function)
        if(rc == EAI_INPROGRESS){

occuring with a glibc-2.27-based buildroot toolchain for sparc64

Target: sparc64-buildroot-linux-gnu
[...]
gcc version 6.4.0 (Buildroot 2018.05)

Source:
http://autobuild.buildroot.org/toolchains/tarballs/br-sparc64-full-2018.05.tar.bz2

Patch sent upstream as PR 933.

Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>
---
 config.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/config.h b/config.h
index 7607019..ba0ba93 100644
--- a/config.h
+++ b/config.h
@@ -39,4 +39,6 @@
 #  define _POSIX_C_SOURCE 200809L
 #endif
 
+#define _GNU_SOURCE
+
 #endif
-- 
2.18.0

